types/src/v0/impls/l1.rs

```diff
@@ -172,6 +173,40 @@ impl L1Client {
         self.wait_for_block(number).await
     }
 
+    /// Get information about the first finalized block with timestamp greater than or equal
+    /// `timestamp`.
+    pub async fn wait_for_finalized_block_with_timestamp(&self, timestamp: U256) -> L1BlockInfo {
+        let interval = self.provider.get_interval();
+
+        // Sleep until approximately the right time for the desired block to appear.
+        let now = U256::from(OffsetDateTime::now_utc().unix_timestamp() as u64);
+        if timestamp > now {
```

Comment:
Suggest to add a bit of info logging to this function otherwise it might sleep and or retry for a long time without logging anything.